packages/sitecore-jss-dev-tools/src/templating/utils.test.ts

```diff
@@ -0,0 +1,179 @@
+      const path = 'C:/Windows';
+      const childFileName = 'child.tsx';
+      const callbackStub = sinon.stub();
+      const input = {
+        path: path,
+        resolveItem: (_: any, name: string) => {
+          return name;
+        },
+        cb: callbackStub,
+      };
+      const parentDir = {
+        ...baseDirent,
+        isDirectory: () => true,
+        name: 'parent',
+      };
+      const childFile = {
+        ...baseDirent,
+        isFile: () => true,
+        name: childFileName,
+      };
+      sinon.stub(fs, 'existsSync').returns(true);
+      const readDirStub = sinon.stub(fs, 'readdirSync');
+      readDirStub.withArgs(path, { withFileTypes: true }).returns([parentDir]);
+      readDirStub.withArgs(`${path}/parent`, { withFileTypes: true }).returns([childFile]);
+      getItems(input);
+
+      expect(callbackStub.calledWith('parent')).to.equal(false);
+      expect(callbackStub.calledWith('child')).to.equal(true);
+    });
+  });
+
+  describe('scaffoldFile', () => {
+    afterEach(() => {
+      sinon.restore();
+    });
+
+    it('should scaffold file', () => {
+      sinon.stub(fs, 'existsSync').returns(false);
+      const mkDirStub = sinon.stub(fs, 'mkdir');
+      const writeFileStub = sinon.stub(fs, 'writeFileSync');
+      const scaffoldInput = {
+        rootPath: 'C:/myapp',
+        fileContent: 'sample content',
+        filename: 'index.ts',
+        componentPath: 'FreshComponent',
+      };
+      const expectedOutputDir = `${scaffoldInput.rootPath}/${scaffoldInput.componentPath}`;
+      const result = scaffoldFile(
+        scaffoldInput.rootPath,
+        scaffoldInput.fileContent,
+        scaffoldInput.filename,
+        scaffoldInput.componentPath
+      );
+      expect(mkDirStub.calledWith(`${expectedOutputDir}`, { recursive: true }));
+      expect(
+        writeFileStub.calledWith(
+          `${expectedOutputDir}/${scaffoldInput.filename}`,
+          editLineEndings(scaffoldInput.fileContent),
+          'utf8'
+        )
+      );
+      expect(result).to.equal('C:\\myapp\\FreshComponent\\index.ts');
+    });
+
+    it('should return null when output file already exists', () => {
+      const scaffoldInput = {
+        rootPath: 'C:/myapp',
+        fileContent: 'sample content',
+        filename: 'index.ts',
+        componentPath: 'FreshComponent',
+      };
+
+      const existsStub = sinon.stub(fs, 'existsSync');
+      existsStub.withArgs('C:\\myapp\\FreshComponent\\index.ts').returns(true);
+      const mkDirStub = sinon.stub(fs, 'mkdir');
+      const writeFileStub = sinon.stub(fs, 'writeFileSync');
+      const result = scaffoldFile(
+        scaffoldInput.rootPath,
+        scaffoldInput.fileContent,
+        scaffoldInput.filename,
+        scaffoldInput.componentPath
+      );
+      expect(result).to.equal(null);
+      expect(mkDirStub.called).to.equal(false);
+      expect(writeFileStub.called).to.equal(false);
+    });
+  });
+  /*
+  describe('writePlugins', () => {
```

Comment:
_writePlugins_ uses _getPluginList_ that uses _getItems_ and you covered _getItems_ by unit tests.
Firstly, you need to add unit tests for _getPluginList_
after that, you will easily add for _writePlugins_
It's not required to mock functions from utils in this case
you just need to create some shared functions for mocking _getItems_ and just reuse that

Reply:
To address both comments: sinon cannot stub functions from used within a single file, while testing that file. The stub will be created but it will not be hit during assertions (as sinon would stub a different instance of a function).
It's probably possible to fix with further breaking up of the utils file, but we're entering the area of diminishing returns with this, just for the sake of the unit tests. The coverage is good enough for now.

Reply:
If coverage value is still good, I think it's fine to skip in this case, not so critical